### ISSUE TITLE
failed to align contigs

### ISSUE THREAD

**Issue by cbplott** [marbl/canu#582]
Hello,

I am trying to fill in a gap in a chromosome. I am using reads that I mapped to the area using minimap2. Every time I run canu it failes in the same place, alignGFA fails. I have checked the coverage of the data, ~27x after trimming, and the read length of the reads are all 1kb or greater. The most resent command I used is:
```canu -p MINITEST_Chr12 -d test_Chr12 genomeSize=40000 corMinCoverage=0 corMhapSensitivity=high minOverlapLength=500 useGrid=0 maxMemory=40 ovsMethod=sequential maxThreads=8 -pacbio-raw chr12.fasta```

This is the error I get:
```
-- Finished stage 'consensusLoad', reset canuIteration.
----------------------------------------
-- Starting command on Fri Aug  4 16:04:37 2017 with 1535.862 GB free disk space

    cd unitigging
    /home/raid2/LINUXOPT/canu-1.5/Linux-amd64/bin/tgStoreCoverageStat \
      -G ./MINITEST_Chr12.gkpStore \
      -T ./MINITEST_Chr12.ctgStore 2 \
      -s 40000 \
      -o ./MINITEST_Chr12.ctgStore.coverageStat \
    > ./MINITEST_Chr12.ctgStore.coverageStat.err 2>&1

-- Finished on Fri Aug  4 16:04:37 2017 (lickety-split) with 1535.862 GB free disk space
----------------------------------------
-- Finished stage 'consensusAnalyze', reset canuIteration.
----------------------------------------
-- Starting command on Fri Aug  4 16:04:37 2017 with 1535.862 GB free disk space

    cd unitigging/4-unitigger
    ./alignGFA.sh
GFA alignments failed.

-- Finished on Fri Aug  4 16:04:38 2017 (1 second) with 1535.862 GB free disk space
----------------------------------------
ERROR:
ERROR:  Failed with exit code 1.  (rc=256)
ERROR:

ABORT:
ABORT: Canu snapshot v1.5 +96 changes (r8296 7a25061e8df6a77f4c62c0be7751a5385a5c8b30)
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   failed to align contigs.
ABORT:
ABORT: Disk space available:  1535.862 GB
ABORT:
```

**Comment by brianwalenz** [comment]
You're copy of canu is about a month old and missing ~100 commits.  I don't see anything specific to this problem in the commit logs, but vaguely remember fixing a similar crash.  Please update and retry.  It sounds like you're assembling reads that are expected to cover a 50k chunk of the genome.  That would trigger the crash I'm thinking of.

If that still fails, the alignGFA.sh script runs four tasks (three in the latest version).  The output logs are in unitigging/4-unitigger/*aligned*err.  Which one is failing, and what's the error?  

Alternatively, at the end of alignGFA.sh it is checking for four output files.  If you create empty files for these, this step will be skipped.  You probably don't need the GFA's anyway.

**Comment by cbplott** [comment]
Brian,
I updated canu yesterday. The assembly still failed. It looks like it is failing on creating chr12_patch.unitigs.aligned.bed. The error log "chr12_patch.unitigs.aligned.bed.err" has the following:
'''-- Reading BED './chr12_patch.unitigs.bed'.
Failed to open './chr12_patch.unitigs.bed' for reading: No such file or directory'''
I created the missing bed file and reran the command with the -assemble option and in ran to completion although I did not get any contigs just 19 unassembled reads.

**Comment by brianwalenz** [comment]
Ah, yes, no contigs is a problem.

To confirm: "27x after trimming" is after Canu's trimming?

You're probably getting caught by suspicious read filtering at the start of the unitigger.  There are five filter algorithms: 'suspicious', 'higherror', 'lopsided', 'spur' and 'deadends'.  I'm going to guess that turning off all but higherror will fix things.

`batOptions="-nofilter suspicious,lopsided,spur,deadends"`

If you can share the reads, I'd be happy to take a look to see what's really going on.

**Comment by cbplott** [comment]
Brian,

I was mistaken with the 27x coverage. That was for a different assembly I
tried using a smaller genome size. With the genome size of 40KB the
coverages are:

-- In gatekeeper store 'correction/chr12_patch.gkpStore':
--   Found 30 reads.
--   Found 293979 bases (7.34 times coverage).
--
-- In gatekeeper store 'trimming/chr12_patch.gkpStore':
--   Found 29 reads.
--   Found 282335 bases (7.05 times coverage).
--
-- In gatekeeper store 'unitigging/chr12_patch.gkpStore':
--   Found 27 reads.
--   Found 162256 bases (4.05 times coverage).

When I used batOptions the assembly ran to completion and it produced a
contig 13KB long. Thank you for your help. I have attached a copy of the
reads
